### ISSUE TITLE
When loading /relations, synapse doesn’t seem to take the direction parameter into account

### ISSUE THREAD

**Issue by justjanne** [matrix-org/synapse#13793]
### Description

I’m trying to implement a client-side PoC for an improved way of loading threads, which would use synapse’s MSC3715 [implementation](https://github.com/matrix-org/synapse/pull/11941).

### Steps to reproduce

- create thread with many events, e.g. 26 events containing each a-z
- make request to `/_matrix/client/unstable/rooms/$roomId/relations/$rootEventId/m.thread?org.matrix.msc3715.dir=f&limit=5` and to
`/_matrix/client/unstable/rooms/$roomId/relations/$rootEventId/m.thread?org.matrix.msc3715.dir=b&limit=5`
- both times the events for z, y, x, w, v are returned

### Homeserver

Delight Team’s Threads Dev Server

### Synapse Version

1.65.0rc2

### Anything else that would be useful to know?

If more information is needed on the environment of the homeserver, please ask @clokep.

**Comment by clokep** [comment]
I was able to reproduce this on the given server, but we weirdly have a test for exactly this that works:

https://github.com/matrix-org/synapse/blob/9772e362aaef0740039662e3dc93a0de1411eb06/tests/rest/client/test_relations.py#L770-L789

Clearly we're missing something, but I'm not sure what. I attempted to modify the unit test to reproduce this by:

1. Specifying `org.matrix.msc3715.dir=f` (instead of letting it fallback to that value).
2. Specifying that we're interested in `m.annotation` specifically (instead of all relations).

I couldn't think of much else to try, but will need to look deeper into this.
